src/tss2-tcti/tctildr.c

```diff
@@ -460,14 +460,34 @@ TSS2_RC
 Tss2_TctiLdr_Initialize (const char *nameConf,
                          TSS2_TCTI_CONTEXT **tctiContext)
 {
-    char name [PATH_MAX] = { 0, }, conf [PATH_MAX] = { 0, };
+    char *name = NULL;
+    char *conf = NULL;
     TSS2_RC rc;
+    size_t combined_length;
 
     if (nameConf == NULL) {
         return Tss2_TctiLdr_Initialize_Ex (NULL, NULL, tctiContext);
     }
+
+    combined_length = strlen (nameConf);
+    if (combined_length > PATH_MAX - 1) {
+        LOG_ERROR ("combined conf length must be between 0 and PATH_MAX");
+        return TSS2_TCTI_RC_BAD_VALUE;
+    }
+    name = calloc(combined_length + 1, sizeof(char));
+    conf = calloc(combined_length + 1, sizeof(char));
     rc = tctildr_conf_parse (nameConf, name, conf);
     if (rc != TSS2_RC_SUCCESS)
-        return rc;
-    return Tss2_TctiLdr_Initialize_Ex (name, conf, tctiContext);
+        goto out;
+    rc = Tss2_TctiLdr_Initialize_Ex (name, conf, tctiContext);
+out:
+    if(name != NULL) {
```

Comment:
SAFE_FREE() from aux_util.h here and below.

Reply:
> SAFE_FREE() from aux_util.h here and below.

Thanks for your review. I've revised the code according to your comments.